### PR TITLE
feat: new logger interface

### PR DESCRIPTION
**Pull request by johnxnguyen**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

We are using Datadog for logging in internal (i.e non production) builds, however there is not an easy way to log via Datadog yet from any framework since Datadog is imported at the root project.

In this PR: 
- Introduce a new `LoggingProtocol` to describe some basic methods for logging at different levels.
- Introduce `Logging` enum that contains an optional provider conforming to `LoggingProtocol`, which will perform the actual logging.
- Make `Logging` also conform to `LoggingProtocol` to provide the minimum friction to create a log. 

How to use it:
- The provider should be set at app launch from the root project.
- Create a log anywhere via the `Logging` enum, e.g  `Logging.info("hello")`
- Pass in some attributes to the log, which can contain any useful metadata.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Source/Logger.swift [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+public extension LoggerProtocol {
+
+  func debug(
+    tag: String,
+    _ message: String,
+    attributes: LogAttributes? = nil
+  ) {
+    var attrs = attributes ?? .init()
+    attrs["tag"] = tag
```
Comment (netbe) [inline Source/Logger.swift:108]: what is the tag for ? any exemple?

**Review by netbe — APPROVED** [review]
LGTM;) I guess we can create something that conforms to LoggerProtocol and contains multiple Loggers: Datadog + oslog for example

**Review by agisilaos — APPROVED** [review]

File: Source/Logger.swift (hunk 1 above)
Comment (johnxnguyen) [reply Source/Logger.swift:108]: I thought it'd be helpful to be able to group logs to specific flow, which is kind of how we have logs now, e.g "network", "calling", etc...

**Comment by johnxnguyen** [comment]
> LGTM;) I guess we can create something that conforms to LoggerProtocol and contains multiple Loggers: Datadog + oslog for example

Yes I agree

**Comment by johnxnguyen** [comment]
release minor

**Pull request merged**